pkg/segment/query/processor/sortcommand.go

```diff
@@ -139,19 +139,33 @@ func compareString(a, b string) compare {
 	return GREATER
 }
 
-func getRank(CValEnc *segutils.CValueEnclosure) dTypeRank {
+func getRank(CValEnc *segutils.CValueEnclosure, op string) dTypeRank {
 	switch CValEnc.Dtype {
 	case segutils.SS_DT_BACKFILL, segutils.SS_INVALID:
 		return OTHER
 	case segutils.SS_DT_SIGNED_NUM, segutils.SS_DT_UNSIGNED_NUM, segutils.SS_DT_FLOAT:
-		return NUMERIC
-	case segutils.SS_DT_STRING:
-		_, err := strconv.ParseFloat(CValEnc.CVal.(string), 64)
-		if err == nil {
-			// If floatValue is possible then it is considered as a number
+		switch op {
+		case "num", "auto", "":
+			return NUMERIC
+		case "str":
+			return STRING
```

Comment:
calling this enum as `STRING` is too generic, we should typically prefix some descriptor for such enums. for e.g. we call the `SS_DT_STRING` (sigscalr datatype string). 
Similarly we could call these things `RANK_STRING` etc.

Reply:
addressed in the latest commit.